utils/patternGenerator.ts

```diff
@@ -0,0 +1,141 @@
+import parseYaml from "../utils/parseYaml";
+
+function getType($ref) {
+  const $refDirName = $ref.split("/");
+  return $refDirName[$refDirName.length - 1];
+}
+
+function getRpcDefinition(rpcDefinitionPath, rpcName) {
+  return parseYaml(rpcDefinitionPath).find(item => item.name === rpcName);
+}
+
+function getSchema() {
+  const baseTypes = parseYaml("../execution-apis/src/schemas/base-types.yaml");
+  const block = parseYaml("../execution-apis/src/schemas/block.yaml");
+  const client = parseYaml("../execution-apis/src/schemas/client.yaml");
+  const filter = parseYaml("../execution-apis/src/schemas/filter.yaml");
+  const receipt = parseYaml("../execution-apis/src/schemas/receipt.yaml");
+  const state = parseYaml("../execution-apis/src/schemas/state.yaml");
+  const transaction = parseYaml("../execution-apis/src/schemas/transaction.yaml");
+  const withdrawal = parseYaml("../execution-apis/src/schemas/withdrawal.yaml");
+
+  return {
+    ...baseTypes,
+    ...block,
+    ...client,
+    ...filter,
+    ...receipt,
+    ...state,
+    ...transaction,
+    ...withdrawal,
+  }
+}
+
+function isMainType(type) {
+  return type.charAt(0) === type.charAt(0).toUpperCase();
```

Comment:
What are you considering main types?
I feel like it would be better if we listed them

Reply:
so main types are ones that starts with a capital letter which consists of base type schemas, main types, or both. for example the `Block` from https://github.com/ethereum/execution-apis/blob/main/src/schemas/block.yaml which unpacks to having a list of properties like:

```
logsBloom: bytes256,
mixHash: hash32,
totalDifficulty: uint
transactions: [TransactionSigned],
withdrawals: [Withdrawal],
...
``` 

basically this check looks at the rpc method to figure out if it is a base type (one of  https://github.com/ethereum/execution-apis/blob/main/src/schemas/base-types.yaml which fetches the pattern directly) or if it is a main type (e.g. `Block` from https://github.com/ethereum/execution-apis/blob/main/src/schemas/block.yaml which makes another look up to unpack its properties)

Reply:
also I think adding a dictionary of the list of main types to check for example:

```
const mainTypes = [
    "AccessList",
    "AccessListEntry",
    "AccountProof",
    "BadBlock",
    "Block",
    "BlockNumberOrTag",
    "BlockNumberOrTagOrHash",
    "BlockTag",
    "Filter",
    "FilterResults",
    "FilterTopic",
    "FilterTopics",
    "GenericTransaction",
    "Log",
    "ReceiptInfo",
    "StorageProof",
    "SyncingStatus",
    "Transaction1559Signed",
    "Transaction2930Signed",
    "Transaction1559Unsigned",
    "Transaction2930Unsigned",
    "TransactionInfo",
    "TransactionLegacySigned",
    "TransactionLegacyUnsigned",
    "TransactionSigned",
    "TransactionUnsigned",
    "Withdrawal",
];
```

would add to the list of things we need to keep an eye out in the event that Ethereum makes some changes to the specification and add new ones. I think that's one of the things we are trying to avoid and instead just directly get the information from them, this is @voyenavion suggestion and why I wrote it to now completely rely on the submodule.

Reply:
I understand now. I think it would be more clear if we had `isBaseType()` checking against https://github.com/ethereum/execution-apis/blob/main/src/schemas/base-types.yaml and checked if it is a basetype or not basetype

Reply:
implemented `isBaseType` here: https://github.com/HorizenOfficial/rpc-tests/pull/2/commits/be62eb60c95d2890a4f993eafc886275c38bad9e